docs/developer_guide.md

```diff
@@ -1,28 +1,51 @@
 # Developer Guide
 
 ## Environment
-Follow the installation instructions in the section [Install standalone application directly from the source](https://github.com/lyft/amundsenfrontendlibrary#install-standalone-application-directly-from-the-spource).
 
-Install the javascript development requirements:
-```bash
-# in ~/<your-path-to-cloned-repo>/amundsenfrontendlibrary/amundsen_application
-$ cd static
-$ npm install --only=dev
-```
 
-To test local changes to the javascript static files:
+### Python
+
+The instructions in [Install standalone application directly from the source](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/installation.md#install-standalone-application-directly-from-the-source) are for running the application as-is. If you wish instead to develop against the python code, the recommended method of installation is as follows:
+
 ```bash
-# in ~/<your-path-to-cloned-repo>/amundsenfrontendlibrary/amundsen_application
-$ cd static
-$ npm run dev-build # builds the development bundle
+# Clone repo
+$ git clone https://github.com/lyft/amundsenfrontendlibrary.git
+
+$ cd amundsenfrontendlibrary
+# Install python resources
+$ python3 -m venv venv
+$ source venv/bin/activate
+(venv) $ pip3 install -r requirements3.txt
+(venv) $ python3 setup.py develop
+
+# Start server
+(venv) $ FLASK_ENV=development python3 amundsen_application/wsgi.py
+# visit http://localhost:5000 to confirm the application is running
 ```
 
-To test local changes to the python files, re-run the wsgi:
+The difference is subtle but significant - instead of running `python3 setup.py install`, which copies the files from the application to the relevant locations, we are using `python3 setup.py develop` which will only link the files - meaning you can continue to edit the files python files in `~/<your-path-to-cloned-repo>/amundsenfrontendlibrary/amundsen_application`.
```

Comment:
Just curious-- I only run `python3 setup.py install` and run my server with `DEBUG=True` in my `LocalConfig`. I haven't encountered any problems with hot reloading of python code while my server is running. I wonder what's different about our setups.

Reply:
@danwom That's interesting - it's not how it works for me. If I run:

```bash
# Clone repo
$ git clone https://github.com/lyft/amundsenfrontendlibrary.git

$ cd amundsenfrontendlibrary
# Install python resources
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip3 install -r requirements3.txt
(venv) $ python3 setup.py install

# Start server
(venv) $ python3 amundsen_application/wsgi.py
# visit http://localhost:5000 to confirm the application is running
```
That is - `python3 setup.py install` - then the flask server will look for changes to the file in the installed package (e.g. ~/amundsenfrontendlibrary/venv/lib/python3.6/site-packages/amundsen_frontend-1.0.7-py3.6.egg/amundsen_application/api/v0.py ) but it's not useful for me to have flask listen to *that* file because it's an artifact of the egg and it's in my venv directory. Instead, if I run:

```bash
# Clone repo
$ git clone https://github.com/lyft/amundsenfrontendlibrary.git

$ cd amundsenfrontendlibrary
# Install python resources
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip3 install -r requirements3.txt
(venv) $ python3 setup.py develop

# Start server
(venv) $ python3 amundsen_application/wsgi.py
# visit http://localhost:5000 to confirm the application is running
```

Then I can edit the file in `~/amundsenfrontendlibrary/amundsen_application/api/v0.py` with any editor of choice and flask will detect that change in real time and serve that file, so then I don't have to run `python3 setup.py install` every time I make a change to a python file.

I learned this method from here: https://stackoverflow.com/questions/19048732/python-setup-py-develop-vs-install

I will be pushing corrections for your other comments.

Reply:
The current setup instructions with default setup works exactly as you described in the first case. 

Perhaps as an oversight the `DEBUG` config has been set to `False` under `LocalConfig`. if you edit [the debug config](https://github.com/lyft/amundsenfrontendlibrary/blob/675a29dedef501c095bbbadb9056d1f6571b82be/amundsen_application/config.py#L23) from `False` to `True`, then the deployment would use your local files that can be edited in real time as described in your second case.

I am suggesting that using config files to get hot reloading might be better than requiring a different set of setup commands, as the configs will give you this behavior by default.